Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/saver/ImageSaveTask.java

```diff
@@ -119,35 +121,41 @@ public void onEnd() {
     }
 
     private void deleteDestination() {
-        if (destination.exists()) {
-            if (!destination.delete()) {
+        if (fileManager.exists(destination)) {
+            if (!fileManager.delete(destination)) {
                 Logger.e(TAG, "Could not delete destination after an interrupt");
             }
         }
     }
 
     private void onDestination() {
         success = true;
-        MediaScannerConnection.scanFile(getAppContext(), new String[]{destination.getAbsolutePath()}, null, (path, uri) -> {
-            // Runs on a binder thread
-            AndroidUtils.runOnUiThread(() -> afterScan(uri));
-        });
+//        String[] paths = {destination.getFullPath()};
+
+        // FIXME: does not work. Who in their right mind even wants their downloaded images
```

Comment:
MediaScanner is a system level service that doesn't necessarily get touched by any Google program, assuming you have a ROM with all the Google stuff removed. It's used for interop between different apps and a computer, assuming you're connected using MTP. Basically, it's not useless and should be reimplemented at some point in time.